### PR TITLE
[V1] fix #509 - php's parse str does not allow dots

### LINKED ISSUE #509: Filter parameter names are corrupted when using nested properties
(This bug applies to both 1.x and master)

Moved from api-platform/api-platform#76

*@vincentchalamon on Apr 21, 2016, 8:51 AM UTC:*

When I call `/reservations?meal.startedAt[after]=2015-01-02+14:00:00` (so with filter `meal.startedAt`), this filter is corrupted on `hydra:firstPage` & `hydra:lastPage` JSON-LD response keys:

```json
{
    "@context": "/contexts/Reservation",
    "@id": "/reservations?meal.startedAt[after]=2015-01-02+14:00:00",
    "@type": "hydra:PagedCollection",
    "hydra:totalItems": 2,
    "hydra:itemsPerPage": 30,
    "hydra:firstPage": "/reservations?meal_startedAt%5Bafter%5D=2015-01-02+14%3A00%3A00",
    "hydra:lastPage": "/reservations?meal_startedAt%5Bafter%5D=2015-01-02+14%3A00%3A00",
    "hydra:member": []
}
```

As you can see, `meal.startedAt` filter is converted to `meal_startedAt`, which is invalid.

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #509
| License       | MIT
| Doc PR        |

Fix the dots to underscore

### CODE REVIEW

**Comment by Simperfit** [comment]
ping @teohhanhui

File: features/doctrine/date-filter.feature [hunk 1]
```diff
@@ -294,3 +294,146 @@ Feature: Order filter on collections
...
+                {
+                    "@type": "IriTemplateMapping",
+                    "variable": "relatedDummy.dummyDate[after]",
+                    "property": "relatedDummy.dummyDate",
+                    "required": false
+                }
+            ]
+        }
+    }
+    """
```
Comment (dunglas) [inline features/doctrine/date-filter.feature]: Can you add an empty line at the end?

**Comment by dunglas** [comment]
:+1: (I just left a small comment)

But after thinking more about it, I really don't like this class `abstract` with static methods. I would prefer an utility class (not `abstract` and without `static` methods. Or a trait.

File: features/doctrine/date-filter.feature (hunk 1 above)
Comment (Simperfit) [inline features/doctrine/date-filter.feature]: added :)

**Comment by Simperfit** [comment]
@dunglas Do you want to rework that class on both branches or in V2 ?

**Comment by dunglas** [comment]
Only in v2 (in v1 it will be BC break).

**Comment by dunglas** [comment]
Thank you @Simperfit!

**Pull request merged**